app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,74 @@
+module Products
+
+  class UserNoteMode
+
+    VALID_MODES = %w(hidden optional required).freeze
+
+    attr_reader :raw_value
+
+    # An array of UserNoteModes representing the possible modes
+    # Use this in validations and form collections.
+    def self.all
+      VALID_MODES.map { |value| new(value) }
+    end
+
+    def self.[](mode)
+      if mode.is_a?(self)
+        mode
+      elsif VALID_MODES.include?(mode)
+        new(mode)
+      else
+        InvalidUserNoteMode.new(mode)
+      end
+    end
+
+    # Prefer the use of the factory `UserNoteMode[raw_value]` over directly
+    # initializing this class with `new` as it will not hard error on invalid
+    # inputs.
+    def initialize(raw_value)
+      @raw_value = raw_value
+      raise ArgumentError, "Invalid value: #{raw_value}" unless VALID_MODES.include?(raw_value)
+      freeze
+    end
+
```

Comment:
Took me a second to get what you were saying. I read (because I wrote it) with the factory being the antecedent of "it". Clearly, it's unclear. I'll try to reword it.